cibuild.sh

```diff
@@ -97,34 +97,34 @@ fi
 
 # obtain_dotnet.sh puts the right dotnet on the PATH
 FORCE_DOWNLOAD=true
-source ${ROOT_PATH}/build/scripts/obtain_dotnet.sh
+source "${ROOT_PATH}"/build/scripts/obtain_dotnet.sh
 
-RUNTIME_ID=$(dotnet --info | awk '/RID:/{print $2;}')
+RUNTIME_ID="$(dotnet --info | awk '/RID:/{print $2;}')"
 echo "Using Runtime Identifier: ${RUNTIME_ID}"
 
 RESTORE_ARGS="-r ${RUNTIME_ID} -v Minimal --disable-parallel"
 echo "Restoring BaseToolset.csproj"
-dotnet restore ${RESTORE_ARGS} ${ROOT_PATH}/build/ToolsetPackages/BaseToolset.csproj
+dotnet restore ${RESTORE_ARGS} "${ROOT_PATH}"/build/ToolsetPackages/BaseToolset.csproj
 echo "Restoring CoreToolset.csproj"
-dotnet restore ${RESTORE_ARGS} ${ROOT_PATH}/build/ToolsetPackages/CoreToolset.csproj
+dotnet restore ${RESTORE_ARGS} "${ROOT_PATH}"/build/ToolsetPackages/CoreToolset.csproj
 echo "Restoring CrossPlatform.sln"
-dotnet restore ${RESTORE_ARGS} ${ROOT_PATH}/CrossPlatform.sln
+dotnet restore ${RESTORE_ARGS} "${ROOT_PATH}"/CrossPlatform.sln
 
 BUILD_ARGS="--no-restore -c ${BUILD_CONFIGURATION} /nologo /consoleloggerparameters:Verbosity=minimal;summary /bl:${BUILD_LOG_PATH} /maxcpucount:1"
 PUBLISH_ARGS="-f ${TARGET_FRAMEWORK} -r ${RUNTIME_ID} ${BUILD_ARGS}"
 
 echo "Building bootstrap csc"
-dotnet publish ${SRC_PATH}/Compilers/CSharp/csc -o ${BOOTSTRAP_PATH}/csc ${PUBLISH_ARGS}
+dotnet publish "${SRC_PATH}"/Compilers/CSharp/csc -o "${BOOTSTRAP_PATH}"/csc ${PUBLISH_ARGS}
 echo "Building bootstrap vbc"
-dotnet publish ${SRC_PATH}/Compilers/VisualBasic/vbc -o ${BOOTSTRAP_PATH}/vbc ${PUBLISH_ARGS}
-rm -rf ${BINARIES_PATH}/${BUILD_CONFIGURATION}
+dotnet publish "${SRC_PATH}"/Compilers/VisualBasic/vbc -o "${BOOTSTRAP_PATH}"/vbc ${PUBLISH_ARGS}
+rm -rf "${BINARIES_PATH:?}"/"${BUILD_CONFIGURATION}"
```

Comment:
Why was the `:?` added here?

For reference to other reviewers - `${parameter:?err_msg}` "[If parameter set, use it, else print err_msg and abort the script with an exit status of 1.](http://www.tldp.org/LDP/abs/html/parameter-substitution.html)" - we already have `set -u`.

(Perhaps you're just being extra careful and guarding against [nuking a computer](https://www.theregister.co.uk/2015/01/17/scary_code_of_the_week_steam_cleans_linux_pcs/)?)

Reply:
Yeah this was a "**don't accidentally** `rm -rf /` **no matter what**" thing.

(I didn't actually notice the `set -u` 😅 )